src/sipmpdf/functions.py

```diff
@@ -259,4 +372,17 @@ def sipm_response(
   Using numba for calculation acceleration (the function is difficult to express
   in pure array syntax)
   """
-  pass
+  kern = kernel_switch(x, pedestal, gain, common_noise, pixel_noise, poisson_mean, poisson_borel, ap_prob, ap_beta)
+ 
+  Pdc=0 ##probability of dark current, define here, fix all these values later
+  smear_of_darkcurrent=0 ##define here
+  resolution_of_darkcurrent=0 ##define here
+  l=0 ##define here  
+  ap_smear=0 ##define here
```

Comment:
This is a dynamic variable that depends on the number of pixel fired: 
`sigma_k = kern.sqrt(sigma_0**2 + k* sigma_1**2)`